tools/nodejs/lib/duckdb.js

```diff
@@ -6,6 +6,23 @@ module.exports = exports = duckdb;
 var Database = duckdb.Database;
 var Connection = duckdb.Connection;
 var Statement = duckdb.Statement;
+var QueryResult = duckdb.QueryResult;
+
+QueryResult.prototype[Symbol.asyncIterator] = async function*() {
+    let prefetch = this.nextChunk();
+    while (true) {
+        const chunk = await prefetch;
+        // Null chunk indicates end of stream
+        if (!chunk) {
+            return;
+        }
+        // Prefetch the next chunk while we're iterating
+        prefetch = this.nextChunk();
```

Comment:
This optimization is worth about a 5% decrease in end-to-end time of the new test I added, so I decided to keep it